File: SpringBoot/demo/src/main/java/com/example/demo/bussines/abstracts/ProductService.java
package com.example.demo.bussines.abstracts;

import com.example.demo.entities.concretes.Product;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface ProductService {
    ResponseEntity<Product> add(Product product);
    ResponseEntity<List<Product>> getAll();
    ResponseEntity<Product> update(int id,Product product);
    ResponseEntity<Product> delete(int id);

    ResponseEntity<Product> getById(int id);

    ResponseEntity<List<Product>> getByCategoryId(int id);
}

File: Angular/demo/src/services/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Product } from 'src/models/product';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  constructor(private http: HttpClient) { }

  getAll(){
    return this.http.get<Product[]>("http://localhost:8080/api/products/getAll")
  }

  getById(id: number){
    return this.http.get<Product>("http://localhost:8080/api/products/getById?id="+ id)
  }

  getByCategoryId(id:number){
    return this.http.get<Product[]>("http://localhost:8080/api/products/getByCategoryId?id="+ id)
  }

  add(value: Object){
    this.http.post("http://localhost:8080/api/products/add",value).subscribe((result)=>{
      console.log(JSON.stringify(result))
    })
  }

  update(id:number, value:Object){
    this.http.post("http://localhost:8080/api/products/update?id="+id,value).subscribe((result)=>{
      console.log(JSON.stringify(result))
    })
  }

  delete(id: number){
    return this.http.delete("http://localhost:8080/api/products/delete?id="+id)
  }
}

